server/src/app.service.ts

```diff
@@ -143,5 +144,59 @@ export class AppService {
         round.roundId,
       );
     }
+
+  }
+  async enterResultsFromExternalDevice(enterResultsDto: EnterResultsDto) {
+    const round = await this.resultsService.getContestRound(
+      enterResultsDto.competitionWcaId,
+      enterResultsDto.eventId,
+      enterResultsDto.roundNumber,
+    );
+
+    for (const externalResultDto of enterResultsDto.results) {
+      const person = externalResultDto.wcaId
+        ? await this.personsService.getPersonByWcaId(externalResultDto.wcaId.toUpperCase())
+        : await this.personsService.getPersonById(externalResultDto.registrantId);
+
+      if (!person) throw new NotFoundException('Competitor not found');
```

Comment:
Considering that this works for multiple results, you might wanna expand this error message to specify which competitor was not found, and include the registrant ID or WCA ID.